### ISSUE TITLE
Misleading success output on MacOS

### ISSUE THREAD

**Issue by strangelittlemonkey** [r-darwish/topgrade#666]
**What did you expect to happen?**
I expected that when run, if a step failed then the summary at the end would state there was a failure in that step.

**What actually happened?**
The tool reported success in the summary.

**Additional details:**
- Which operating system or Linux distribution are you using? MacOS 10.15.7
- How did you install Topgrade? Brew

Run `topgrade --dry-run` to see which commands Topgrade is running. If the command seems wrong and you know why pleaes tell us so.
If the command seems fine try to run it yourself and tell us if you got a different result from Topgrade.

<details><pre>
Paste the output of `topgrade -v`
</pre></details>

 topgrade -v
 2021-03-12T09:02:27.545Z DEBUG topgrade::config > Configuration directory /Users/zach/.config does not exist
 2021-03-12T09:02:27.545Z DEBUG topgrade         > Version: 6.7.0
 2021-03-12T09:02:27.545Z DEBUG topgrade         > OS: x86_64-apple-darwin
 2021-03-12T09:02:27.545Z DEBUG topgrade         > Args { inner: ["topgrade", "-v"] }
 2021-03-12T09:02:27.545Z DEBUG topgrade         > Binary path: Ok("/usr/local/bin/topgrade")
 2021-03-12T09:02:27.545Z DEBUG topgrade         > Self Update: false
 2021-03-12T09:02:27.546Z DEBUG topgrade::utils  > Detected "/usr/bin/git" as "git"
 2021-03-12T09:02:27.546Z DEBUG topgrade::utils  > Cannot find "doas"
 2021-03-12T09:02:27.546Z DEBUG topgrade::utils  > Detected "/usr/bin/sudo" as "sudo"
 2021-03-12T09:02:27.546Z DEBUG topgrade::utils  > Cannot find "pwsh"
 2021-03-12T09:02:27.546Z DEBUG topgrade::utils  > Cannot find "powershell"
 2021-03-12T09:02:27.546Z DEBUG topgrade::runner > Step "Brew (ARM)"
 2021-03-12T09:02:27.546Z DEBUG topgrade::runner > Step "Brew (Intel)"
 2021-03-12T09:02:27.547Z DEBUG topgrade::utils  > Detected "/usr/local/bin/brew" as "/usr/local/bin/brew"

―― 01:02:27 - Brew ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2021-03-12T09:02:27.547Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "update"
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Casks
brave-browser ✔                                                    sublime-merge
 2021-03-12T09:02:42.102Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "upgrade" "--ignore-pinned" "--formula"
 2021-03-12T09:02:44.103Z DEBUG topgrade::runner   > Step "Brew Cask (ARM)"
 2021-03-12T09:02:44.104Z DEBUG topgrade::runner   > Step "Brew Cask (Intel)"
 2021-03-12T09:02:44.104Z DEBUG topgrade::utils    > Detected "/usr/local/bin/brew" as "/usr/local/bin/brew"

―― 01:02:44 - Brew - Cask ――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2021-03-12T09:02:45.611Z DEBUG topgrade::executor > Running "/usr/local/bin/brew" "upgrade" "--cask"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "MacPorts"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "Microsoft AutoUpdate"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "yadm"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "nix"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "home-manager"
 2021-03-12T09:02:47.241Z DEBUG topgrade::runner   > Step "asdf"
 2021-03-12T09:02:47.241Z DEBUG topgrade::utils    > Path "/Users/zach/.config/emacs" doesn't exist
 2021-03-12T09:02:47.241Z DEBUG topgrade::utils    > Path "/Users/zach/.emacs.d" exists
 2021-03-12T09:02:47.241Z DEBUG topgrade::utils    > Path "/Users/zach/.emacs.d/bin/doom" doesn't exist
 2021-03-12T09:02:47.241Z DEBUG topgrade::steps::git > Checking if /Users/zach/.emacs.d is a git repository
 2021-03-12T09:02:47.251Z TRACE topgrade::executor   > Output of "/usr/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(ExitStatus(32768)), stdout: "", stderr: "fatal: not a git repository (or any of the parent directories): .git\n" }
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.doom.d does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.vim does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.config/nvim does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.zshrc does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.tmux does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/.config/fish does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/Library/Application Support/openbox does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/Library/Application Support/bspwm does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/Library/Application Support/i3 does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::steps::git > /Users/zach/Library/Application Support/sway does not exists
 2021-03-12T09:02:47.251Z DEBUG topgrade::runner     > Step "Git repositories"
 2021-03-12T09:02:47.251Z DEBUG topgrade::runner     > Step "zr"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.252Z DEBUG topgrade::runner     > Step "antibody"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.252Z DEBUG topgrade::runner     > Step "antigen"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.252Z DEBUG topgrade::runner     > Step "zplug"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.252Z DEBUG topgrade::runner     > Step "zinit"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.252Z DEBUG topgrade::runner     > Step "zim"
 2021-03-12T09:02:47.252Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.262Z TRACE topgrade::executor   > Output of "zsh" "-c" "[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}": Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "" }
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "oh-my-zsh"
 2021-03-12T09:02:47.262Z DEBUG topgrade::utils      > Detected "/bin/zsh" as "zsh"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "fisher"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "bash-it"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "oh-my-fish"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "tmux"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "TLDR"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "pearl"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "SDKMAN!"
 2021-03-12T09:02:47.262Z DEBUG topgrade::utils      > Detected "/bin/bash" as "bash"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "apm"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "vscode"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "vscodium"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "fossil"
 2021-03-12T09:02:47.262Z DEBUG topgrade::runner     > Step "rustup"
 2021-03-12T09:02:47.263Z DEBUG topgrade::utils      > Detected "/Users/zach/.cargo/bin/rustup" as "rustup"

―― 01:02:47 - rustup ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2021-03-12T09:02:47.263Z DEBUG topgrade::executor   > Running "/Users/zach/.cargo/bin/rustup" "self" "update"
info: checking for self-updates
  rustup unchanged - 1.23.1

 2021-03-12T09:02:47.447Z DEBUG topgrade::executor   > Running "/Users/zach/.cargo/bin/rustup" "update"
info: syncing channel updates for 'stable-x86_64-apple-darwin'
info: checking for self-updates

  stable-x86_64-apple-darwin unchanged - rustc 1.50.0 (cb75ad5db 2021-02-10)

info: cleaning up downloads & tmp directories
 2021-03-12T09:02:47.787Z DEBUG topgrade::runner     > Step ".NET"
 2021-03-12T09:02:47.787Z DEBUG topgrade::runner     > Step "choosenim"
 2021-03-12T09:02:47.787Z DEBUG topgrade::runner     > Step "cargo"
 2021-03-12T09:02:47.787Z DEBUG topgrade::utils      > Detected "/Users/zach/.cargo/bin/cargo" as "cargo"

―― 01:02:47 - Cargo ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2021-03-12T09:02:47.787Z DEBUG topgrade::utils      > Detected "/Users/zach/.cargo/bin/cargo-install-update" as "cargo-install-update"
 2021-03-12T09:02:47.787Z DEBUG topgrade::executor   > Running "/Users/zach/.cargo/bin/cargo-install-update" "install-update" "--git" "--all"
    Updating registry 'https://github.com/rust-lang/crates.io-index'

Package       Installed  Latest  Needs update
cargo-update  v5.0.0     v5.0.0  No

No packages need updating.
Package  Installed  Latest  Needs update

No git packages need updating.
Overall updated 0 packages.
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "Flutter"
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "Emacs"
 2021-03-12T09:02:48.751Z DEBUG topgrade::utils      > Detected "/usr/local/bin/emacs" as "emacs"
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "opam"
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "vcpkg"
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "pipx"
 2021-03-12T09:02:48.751Z DEBUG topgrade::runner     > Step "stack"
 2021-03-12T09:02:48.752Z DEBUG topgrade::runner     > Step "tlmgr"
 2021-03-12T09:02:48.752Z DEBUG topgrade::runner     > Step "myrepos"
 2021-03-12T09:02:48.752Z DEBUG topgrade::runner     > Step "jetpack"
 2021-03-12T09:02:48.752Z DEBUG topgrade::runner     > Step "vim"
 2021-03-12T09:02:48.752Z DEBUG topgrade::utils      > Detected "/usr/bin/vim" as "vim"
 2021-03-12T09:02:48.775Z TRACE topgrade::executor   > Output of "/usr/bin/vim" "--version": Output { status: ExitStatus(ExitStatus(0)), stdout: "VIM - Vi IMproved 8.1 (2018 May 18, compiled Dec 10 2020 20:32:49)\nmacOS version\nIncluded patches: 1-503, 505-680, 682-2292\nCompiled by root@apple.com\nNormal version without GUI.  Features included (+) or not (-):\n+acl               -farsi             -mouse_sysmouse    -tag_any_white\n-arabic            +file_in_path      -mouse_urxvt       -tcl\n+autocmd           +find_in_path      +mouse_xterm       -termguicolors\n+autochdir         +float             +multi_byte        +terminal\n-autoservername    +folding           +multi_lang        +terminfo\n-balloon_eval      -footer            -mzscheme          +termresponse\n-balloon_eval_term +fork()            +netbeans_intg     +textobjects\n-browse            -gettext           +num64             +textprop\n+builtin_terms     -hangul_input      +packages          +timers\n+byte_offset       +iconv             +path_extra        +title\n+channel           +insert_expand     -perl              -toolbar\n+cindent           +job               +persistent_undo   +user_commands\n-clientserver      +jumplist          +postscript        -vartabs\n+clipboard         -keymap            +printer           +vertsplit\n+cmdline_compl     +lambda            -profile           +virtualedit\n+cmdline_hist      -langmap           +python/dyn        +visual\n+cmdline_info      +libcall           -python3           +visualextra\n+comments          +linebreak         +quickfix          +viminfo\n-conceal           +lispindent        +reltime           +vreplace\n+cryptv            +listcmds          -rightleft         +wildignore\n+cscope            +localmap          +ruby/dyn          +wildmenu\n+cursorbind        -lua               +scrollbind        +windows\n+cursorshape       +menu              +signs             +writebackup\n+dialog_con        +mksession         +smartindent       -X11\n+diff              +modify_fname      -sound             -xfontset\n+digraphs          +mouse             +spell             -xim\n-dnd               -mouseshape        +startuptime       -xpm\n-ebcdic            -mouse_dec         +statusline        -xsmp\n-emacs_tags        -mouse_gpm         -sun_workshop      -xterm_clipboard\n+eval              -mouse_jsbterm     +syntax            -xterm_save\n+ex_extra          -mouse_netterm     +tag_binary        \n+extra_search      +mouse_sgr         -tag_old_static    \n   system vimrc file: \"$VIM/vimrc\"\n     user vimrc file: \"$HOME/.vimrc\"\n 2nd user vimrc file: \"~/.vim/vimrc\"\n      user exrc file: \"$HOME/.exrc\"\n       defaults file: \"$VIMRUNTIME/defaults.vim\"\n  fall-back for $VIM: \"/usr/share/vim\"\nCompilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      \nLinking: gcc   -L/usr/local/lib -o vim        -lm -lncurses  -liconv -framework Cocoa           \n", stderr: "" }
 2021-03-12T09:02:48.775Z DEBUG topgrade::runner     > Step "Neovim"
 2021-03-12T09:02:48.775Z DEBUG topgrade::runner     > Step "voom"
 2021-03-12T09:02:48.775Z DEBUG topgrade::runner     > Step "npm"
 2021-03-12T09:02:48.775Z DEBUG topgrade::runner     > Step "yarn"
 2021-03-12T09:02:48.775Z DEBUG topgrade::runner     > Step "deno"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "composer"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "krew"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "gem"
 2021-03-12T09:02:48.776Z DEBUG topgrade::utils      > Detected "/usr/bin/gem" as "gem"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "sheldon"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "rtcl"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "gcloud"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "micro"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "App Store"
 2021-03-12T09:02:48.776Z DEBUG topgrade::runner     > Step "System upgrade"

―― 01:02:48 - macOS system update ――――――――――――――――――――――――――――――――――――――――――――――
Finding available software
 2021-03-12T09:02:56.662Z DEBUG topgrade::steps::os::macos > Output { status: ExitStatus(ExitStatus(0)), stdout: "Software Update Tool\n\nFinding available software\nSoftware Update found the following new or updated software:\n* Label: Safari14.0.3CatalinaAuto-14.0.3\n\tTitle: Safari, Version: 14.0.3, Size: 81419K, Recommended: YES, \n", stderr: "" }
 2021-03-12T09:02:56.662Z DEBUG topgrade::steps::os::macos > ""
A system update is available. Do you wish to install it? (y)es/(N)o
 2021-03-12T09:03:03.869Z DEBUG topgrade::executor         > Running "softwareupdate" "--install" "--all" "--no-scan"
Software Update Tool

Finding available software
Attempting to quit apps: (
    "com.apple.Safari.SafariQuickLookPreview",
    "com.apple.Safari"
)
Waiting for user to quit any relevant apps
Successfully quit all apps

Downloading Safari
Error downloading Safari: The operation couldn’t be completed. (PKDownloadError error 8.)
Done.

Error downloading updates.

―― 01:03:15 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Brew (ARM): SKIPPED: Cannot find "/opt/homebrew/bin/brew" in PATH
Brew (Intel): OK
Brew Cask (ARM): SKIPPED: Cannot find "/opt/homebrew/bin/brew" in PATH
Brew Cask (Intel): OK
MacPorts: SKIPPED: Cannot find "port" in PATH
Microsoft AutoUpdate: SKIPPED: Path "/Library/Application Support/Microsoft/MAU2.0/Microsoft AutoUpdate.app/Contents/MacOS/msupdate" doesn't exist
yadm: SKIPPED: Cannot find "yadm" in PATH
nix: SKIPPED: Cannot find "nix" in PATH
home-manager: SKIPPED: Cannot find "home-manager" in PATH
asdf: SKIPPED: Cannot find "asdf" in PATH
Git repositories: SKIPPED: No repositories to pull
zr: SKIPPED: Cannot find "zr" in PATH
antibody: SKIPPED: Cannot find "antibody" in PATH
antigen: SKIPPED: Path "/Users/zach/.zshrc" doesn't exist
zplug: SKIPPED: Path "/Users/zach/.zshrc" doesn't exist
zinit: SKIPPED: Path "/Users/zach/.zshrc" doesn't exist
zim: SKIPPED: Path "/Users/zach/.zim" doesn't exist
oh-my-zsh: SKIPPED: Path "/Users/zach/.oh-my-zsh" doesn't exist
fisher: SKIPPED: Cannot find "fish" in PATH
bash-it: SKIPPED: Path "/Users/zach/.bash_it" doesn't exist
oh-my-fish: SKIPPED: Cannot find "fish" in PATH
tmux: SKIPPED: Path "/Users/zach/.tmux/plugins/tpm/bin/update_plugins" doesn't exist
TLDR: SKIPPED: Cannot find "tldr" in PATH
pearl: SKIPPED: Cannot find "pearl" in PATH
SDKMAN!: SKIPPED: Path "/Users/zach/.sdkman/bin/sdkman-init.sh" doesn't exist
apm: SKIPPED: Cannot find "apm" in PATH
vscode: SKIPPED: Cannot find "code" in PATH
vscodium: SKIPPED: Cannot find "codium" in PATH
fossil: SKIPPED: Cannot find "fossil" in PATH
rustup: OK
.NET: SKIPPED: Cannot find "dotnet" in PATH
choosenim: SKIPPED: Cannot find "choosenim" in PATH
cargo: OK
Flutter: SKIPPED: Cannot find "flutter" in PATH
Emacs: SKIPPED: Path "/Users/zach/.emacs.d/init.el" doesn't exist
opam: SKIPPED: Cannot find "opam" in PATH
vcpkg: SKIPPED: Cannot find "vcpkg" in PATH
pipx: SKIPPED: Cannot find "pipx" in PATH
stack: SKIPPED: Cannot find "stack" in PATH
tlmgr: SKIPPED: Cannot find "tlmgr" in PATH
myrepos: SKIPPED: Cannot find "mr" in PATH
jetpack: SKIPPED: Cannot find "jetpack" in PATH
vim: SKIPPED: Path "/Users/zach/.vim/vimrc" doesn't exist
Neovim: SKIPPED: Cannot find "nvim" in PATH
voom: SKIPPED: Cannot find "voom" in PATH
npm: SKIPPED: Cannot find "npm" in PATH
yarn: SKIPPED: Cannot find "yarn" in PATH
deno: SKIPPED: Cannot find "deno" in PATH
composer: SKIPPED: Cannot find "composer" in PATH
krew: SKIPPED: Cannot find "kubectl-krew" in PATH
gem: SKIPPED: Path "/Users/zach/.gem" doesn't exist
sheldon: SKIPPED: Cannot find "sheldon" in PATH
rtcl: SKIPPED: Cannot find "rupdate" in PATH
gcloud: SKIPPED: Cannot find "gcloud" in PATH
micro: SKIPPED: Cannot find "micro" in PATH
App Store: SKIPPED: Cannot find "mas" in PATH
System upgrade: OK
 2021-03-12T09:03:15.392Z DEBUG topgrade::terminal         > Desktop notification: Topgrade finished sucessfully

**Comment by r-darwish** [comment]
I do [check the exit code](https://github.com/r-darwish/topgrade/blob/31f68791e41f9cd73b0934965fdcf70d2e205de7/src/steps/os/macos.rs#L75) so I think `softwareupdate` returns 0 even on errors. We can parse the output but I'd like to avoid that unless this is a big issue.
